### DISCUSSION TITLE
Iplass support anything like @Transactional ?

### DISCUSSION THREAD

**Discussion by kyowanghwe** [dentsusoken/iPLAss#1654] [Q&A]
I want to rollback the data when exception occur but i dont know how to implement with iplass

**Comment by hkondoh** [comment]
Hi @kyowanghwe,

Thanks for using iPLAss!
I assume that your question is about implementing custom logics.

For transaction control within a custom logic, you can use the `Transaction` interface.
Calling `Transaction#setRollbackOnly()` will mark the transaction for rollback.
If you want to rollback immediately, you can use `Transaction#rollback()` instead.

For more details on the `Transaction` interface, please refer to the following documentation.
https://iplass.org/docs/developerguide/customizing/index.html#Transaction
https://iplass.org/docs/javadoc/org/iplass/mtp/transaction/Transaction.html

(Please note that the documentation is currentlly available only in Japanese. I apologize for the inconvenience, but please use a translation tool for the time being.)

Here is an example that retrieves the current transaction and marks it for rollback:

```java
import org.iplass.mtp.transaction.Transaction;

...

// Retrieve the current transaction of the Command. If no transaction has been started, it will start a new one
Transaction.required(t -> { 
    // Mark for rollback
    t.setRollbackOnly()

    // (Optional) Set a callback to be executed on rollback
    t.afterRollback(() -> {
        System.err.println("Rollback!");
    });
});
```

For Commands executed from Actions, if an exception is thrown from `Command#execute()`, the current transaction of the Command will be rolled back by default.
This rollback behavior can be configured via the Admin Console or the annotations for the Command class.

For more details on configuring Command behavior, please refer to the following documentation:
https://iplass.org/docs/developerguide/customizing/index.html#Action-Command
